src/components/Home/components/PostFeed/PostItem/components/NewComment/components/NewCommentForm.jsx

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from "react";
+
+import NewCommentSubmit from "./NewCommentSubmit";
+
+import TextInput from "@components/TextInput";
+
+import api from "@utilities/api";
+import { checkFormValidity } from "@utilities/object";
+import { contactProps, funcProp, numberProp } from "@utilities/propTypeDefs";
+import {
+  newCommentFormSetup,
+  newCommentInitialForm,
+} from "@utilities/formTemplates";
+
+//TODO: kinda repeating myself here... I expect you to handle this future Edd...
```

Comment:
Whenever you see repetition, the best first step is to try and identify exactly what differences there are. If something is different, that's likely an input. If you can define what the inputs are, you can move functionality into a function and then re-use the function instead of repeating code

Reply:
Noted, a `Form` component of some sort definitely can be prized out of this. The forms and the `TextInput` component turned out to be the most complex bit of the exercise for me so I think I was happy just having it working.

Reply:
You could make use of the `children` prop that automatically gets passed to components to help here. When I'm doing forms, I like to structure them like this:

```jsx
export default function Form({ children, onSubmit }) {
    const handleSubmit = (e) => {
        e.preventDefault()
        onSubmit()
    }

    return (
        <form onSubmit={handleSubmit}>
            {children}
        </form>
    )
}
```

```jsx
export default function TextInput({ name, label, onChange, value, placeholder }) {
    return (
        <>
            <label htmlFor={name}>{label}</label>
            <input type="text" value={value} onChange={onChange} placeholder={placeholder} />
        </>
    )
}
```
^ You can do this for each type of input, e.g. checkbox, or have a generic Input component and use props to decide whether the input is a checkbox or radio etc.

```jsx
export default function App() {
    return (
        <Form>
            <TextInput/>
            <RadioInput/>
        </Form>
    )
}
```